### PR TITLE
feat: search block

### PR DESCRIPTION
**Pull request by andreituicu**
Test URLs:
- Before: https://main--helix-block-collection--adobe.hlx.page/block-collection/search
- After: https://search-block--helix-block-collection--adobe.hlx.page/block-collection/search

### CODE REVIEW

File: blocks/search/search.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
+.search .search-box {
+  display: flex;
```
Comment (fkakatie) [inline blocks/search/search.css:2]:
```suggestion
  display: grid;
  grid-template-columns: auto 1fr;
  gap: 4px;
```

File: blocks/search/search.css [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  align-items: center;
+}
+
+.search .search-box input {
+  width: calc(100% - 30px);
```
Comment (fkakatie) [inline blocks/search/search.css:7]: if we use `grid` (or even `flex`), we can omit this width calculation.
```suggestion
```

File: blocks/search/search.css [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+.search ul > li img {
+  width: 100%;
+  aspect-ratio: 4 / 3;
+  object-fit: cover;
+}
+
+.search.minimal ul {
+  padding-left: 24px;
```
Comment (fkakatie) [inline blocks/search/search.css:59]: changes above to the `grid`/`flex` pattern above would have alignment implications here.

File: blocks/search/search.css (hunk 1 above)
Comment (andreituicu) [reply blocks/search/search.css]: Done. Thank you! 👍

File: blocks/search/search.css (hunk 3 above)
Comment (andreituicu) [reply blocks/search/search.css]: Done. Thank you! 👍

File: blocks/search/search.css [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+.search.minimal .search-results p {
+  margin-top: 0;
+}
+
+.search.minimal .search-results a:any-link {
+  text-decoration: underline;
+  color: var(--link-color);
+}
+
+.search.minimal .search-result p:last-child {
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply blocks/search/search.css]: True, I thought I put the anchor inside of a paragraph too. Done. Thank you! 👍

File: blocks/search/search.css (hunk 2 above)
Comment (andreituicu) [reply blocks/search/search.css]: Done. Thank you! 👍

**Review by fkakatie — APPROVED** [review]

**Pull request merged**